docs/proposals/cross-tenant-query-federation.md

```diff
@@ -0,0 +1,82 @@
+---
+title: "Cross-Tenant Query Federation"
+linkTitle: "Cross-Tenant Query Federation"
+weight: 1
+slug: "cross-tenant-query-federation"
+---
+
+- Author: [Christian Simon](https://github.com/simonswine)
+- Date: October 2020
+- Status: Proposed
+
+## Overview
+
+This document aims to describe how to implement the ability to allow queries to cover data from more than a single Cortex tenant.
+
+## Reasoning
+
+Adopting a tenancy model within an organization with each tenant representing a department comes with the disadvantage that it will prevent queries from spanning multiple departments. This proposal tries to overcome those limitations.
+
+## Alternatives considered
+
+In theory PromQL API clients could be aggregating/correlating query results from multiple tenants. For example Grafana could be used with multiple data sources and a cross tenant query could be achieved through using [Transformations][grafana_transformation].
+
+[grafana_transformation]: https://grafana.com/docs/grafana/latest/panels/transformations/
+
+## Challenges
+
+### Aggregate data without overlaps
+
+#### Challenge
+
+The series in different tenants might have exactly the same labels and hence potentially collide which each other.
+
+#### Proposal
+
+In order to identify the series uniquely, queries using multiple tenants should inject a label with the name `__tenant__` and its value containing the tenant ID into the results. An potentially existing label with the same name should be overwritten.
+
+### Exposure of feature to the user
+
+#### Challenge
+
+The tenant ID is currently read from the `X-Scope-OrgID` HTTP header. There is no limitation on character classes defined, that can be used as part of the tenant ID.
```

Comment:
**thought:** There is no limit on character classes explicitly documented in this repository. However, it is unlikely Cortex would work correctly with certain characters such as `/`. This is because `/` is commonly used as a delimiter in our storage backends and having a tenant ID with this character would likely issues in most of our processes. 

The exact restrictions on the tenant id in Cortex would vary based on the storage backends being used.

Reply:
We know for sure parts of Cortex don't work with `/` and whitespaces in the tenant ID.

Reply:
On the back of this Jacob created https://github.com/cortexproject/cortex/issues/3365, will reference it once we have it documented

Reply:
Thanks for adding the limitations to the documentation @pracucci